packages/core/src/DdSdkReactNativeConfiguration.tsx

```diff
@@ -62,6 +77,123 @@ export class DdSdkReactNativeConfiguration {
         readonly trackInteractions: boolean = false,
         readonly trackResources: boolean = false,
         readonly trackErrors: boolean = false,
-        readonly trackingConsent: TrackingConsent = TrackingConsent.GRANTED
+        readonly trackingConsent: TrackingConsent = DEFAULTS.trackingConsent
     ) {}
 }
+
+export type AutoInstrumentationConfiguration = {
+    readonly trackInteractions: boolean;
+    readonly trackResources: boolean;
+    readonly firstPartyHosts?: string[];
+    readonly resourceTracingSamplingRate?: number;
+    readonly trackErrors: boolean;
+};
+
+export type AutoInstrumentationParameters = {
+    readonly trackInteractions: boolean;
+    readonly trackResources: boolean;
+    readonly firstPartyHosts: string[];
+    readonly resourceTracingSamplingRate: number;
+    readonly trackErrors: boolean;
+};
+
+/**
+ * We could use `Proxy` instead of this function, but `Proxy` is not available on
+ * the older android jsc that can still be used.
+ */
+export const addDefaultValuesToAutoInstrumentationConfiguration = (
+    features: AutoInstrumentationConfiguration
+): AutoInstrumentationParameters => {
+    return {
+        ...features,
+        firstPartyHosts:
+            features.firstPartyHosts || DEFAULTS.getFirstPartyHosts(),
+        resourceTracingSamplingRate:
+            features.resourceTracingSamplingRate ||
+            DEFAULTS.resourceTracingSamplingRate
+    };
+};
+
+export type PartialInitializationConfiguration = {
```

Comment:
so partial means "without any auto-instrumentation"? Isn't it (the name) confusing a bit?

Reply:
Partial initialization is made in two steps:
- first we do the auto-instrumentation when the app starts
- then the user initializes the native SDK

So we need 2 parts for the config: first the auto-instrumentation config, then the config for the native SDK. I thought `NativeSDKConfiguration` would be weird as users don't have to know this is for the Native SDK, so I went with `PartialInitializationConfiguration`. I'm open to other suggestions :)